### PR TITLE
 Feat webpack

### PR DESCRIPTION
**Pull request by mixcas**
# It's finally here!

I think u need to run `yarn` to install all node_components.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -5,13 +5,7 @@
...
   $templateuri = get_template_directory_uri();
 
-  if (WP_DEBUG) {
-    $javascriptLibrary = $templateuri . '/dist/js/library.js';
-    $javascriptMain = $templateuri . '/src/js/main.js';
-  } else {
-    $javascriptLibrary = $templateuri . '/dist/js/library.min.js';
-    $javascriptMain = $templateuri . '/dist/js/main.min.js';
-  }
+  $javascriptMain = $templateuri . '/dist/js/main.js';
```
Comment (varffs) [inline functions.php]: why remove this?

File: src/js/main.js [hunk 2]
```diff
@@ -1,34 +1,39 @@
...
-  init: function() {
-    var _this = this;
+// Import dependencies
+import lazySizes from 'lazysizes';
 
-    $(window).resize(function(){
-      _this.onResize();
-    });
+// Import style
+import '../styl/site.styl';
```
Comment (varffs) [inline src/js/main.js]: ?!

File: src/styl/site.styl [hunk 3]
```diff
@@ -1,7 +1,5 @@
-  // styl property ordering: https://github.com/rossPatton/stylint/blob/master/src/data/ordering.json
-
-@import 'kouto-swiss'
-@import '../../globie-library/helpers'
+@import '~kouto-swiss/index.styl'
```
Comment (varffs) [inline src/styl/site.styl]: how does this work?

File: functions.php (hunk 1 above)
Comment (mixcas) [reply functions.php]: cuz dist/js/main.js is the minified file, theres no min file this time. library is gone cuz is all in main

File: src/styl/site.styl (hunk 3 above)
Comment (mixcas) [reply src/styl/site.styl]: `~` is basically node_modules and other common module paths

File: src/styl/site.styl [hunk 4]
```diff
@@ -1,7 +1,5 @@
...
+@import '~globie-styl-library/helpers'
```
Comment (mixcas) [inline src/styl/site.styl]: noticed this?
https://www.npmjs.com/package/globie-styl-library

Comment (mixcas) [reply src/styl/site.styl]: 😎

File: functions.php (hunk 1 above)
Comment (varffs) [reply functions.php]: does it generate a map?

Comment (mixcas) [reply functions.php]: obvs 😎

**Comment by varffs** [comment]
Can you update the Readme plz for how to use dependencies. Also what happens if (somehow) a dependency isn't on npm?

**Comment by mixcas** [comment]
I will update the Readme.

It's not complicated, we can put the not npm library somewhere in our repo and just `import` it in main. That should work. 

Yeah, I think we have to using it to find out…

**Comment by mikeyrayvon** [comment]
i committed `main.js` with the lazySizes import commented out. this is because `lazySizes.init();` is commented, resulting in this jshint error:

```
WARNING in ./src/js/main.js
(Emitted value instead of an instance of Error) jshint results in errors
  'lazySizes' is defined but never used. @ line 5 char 8
    import lazySizes from 'lazysizes';
```

**Pull request merged**